apps/extension/src/service-worker.ts

```diff
@@ -43,6 +43,15 @@ import { walletIdCtx } from '@penumbra-zone/services/ctx/wallet-id';
 
 import { backOff } from 'exponential-backoff';
 
+import { OffscreenWorker } from '@repo/chrome-offscreen-worker/worker';
+
+globalThis.Worker = OffscreenWorker;
+OffscreenWorker.configure(
+  chrome.runtime.getURL(
+    'offscreen.html',
+  ) as `chrome-extension://${typeof chrome.runtime.id}/${'offscreen.html'}`,
+);
+
```

Comment:
the offscreen worker polyfill is applied to the global here. any packages bundled into this entry calling `new Worker` will actually be calling `new OffscreenWorker`